app/controllers/idv/otp_verification_controller.rb

```diff
@@ -0,0 +1,113 @@
+module Idv
+  class OtpVerificationController < ApplicationController
+    include IdvSession
+    include TwoFactorAuthenticatable
+
+    # Skip TwoFactorAuthenticatable before action that depends on MFA contexts
+    # to redirect to account url for signed in users. This controller does not
+    # use or maintain MFA contexts
+    skip_before_action :check_already_authenticated
+
+    before_action :confirm_two_factor_authenticated
+    before_action :confirm_step_needed
+    before_action :handle_locked_out_user
+    before_action :confirm_otp_delivery_preference_selected
+    before_action :confirm_otp_sent, only: %i[show update]
+    before_action :set_code, only: %i[show update]
+    before_action :set_otp_verification_presenter, only: %i[show update]
+
+    def new
```

Comment:
It gets a touch tricky since you can't redirect via a POST request. That said, we can change it up so the delivery method selection controller sends the initial OTP and this controller is used for resending. Didn't do that here since a GET request is the status quo and I didn't wanna take on too much at once, but I can if we think it is important.

Reply:
Moving the code to send the OTP to a service object will also make this a bit easier

Reply:
yeah, that's what I was thinking. I'll leave it up to you. Eventually, my hope is that we replace the "get another code" link with a link to go back to the delivery method page and choose another way to send the OTP (or try the same method again).

Reply:
Yeah, I'm thinking a service object to send the code that we use in the otp delivery preference controller and then in a new `OtpResendController` will come out a bit neater. Then the `OtpVerification` controller can focus on verifying OTPs.